lib/Driver/Driver.cpp

```diff
@@ -413,9 +413,21 @@ std::unique_ptr<Compilation> Driver::buildCompilation(
   bool ShowIncrementalBuildDecisions =
     ArgList->hasArg(options::OPT_driver_show_incremental);
 
-  bool Incremental = ArgList->hasArg(options::OPT_incremental) &&
-    !ArgList->hasArg(options::OPT_whole_module_optimization) &&
-    !ArgList->hasArg(options::OPT_embed_bitcode);
+  bool Incremental = ArgList->hasArg(options::OPT_incremental);
+  if (ArgList->hasArg(options::OPT_whole_module_optimization)) {
+    if (ShowIncrementalBuildDecisions) {
+      llvm::outs() << "Incremental compilation has been disabled in favor of "
```

Comment:
"in favor of" is sort of a funny way to put it. Usually when these things are errors the message is something like "-incremental is not compatible with -whole-module-optimization". In this case the only reason we don't have a diagnostic is to make it easier on Xcode, which doesn't have a way to turn off incremental mode.